lib/fiware-iotagent-lib.js

```diff
@@ -116,6 +129,19 @@ function doActivate(newConfig, callback) {
 
     logger.info(context, 'Activating IOT Agent NGSI Library.');
 
+    if (newConfig.authentication && newConfig.authentication.enabled) {
+        if (newConfig.authentication.type &&
+            newConfig.authentication.type === 'oauth2') {
+            logger.info(context, 'OAuth2 selected as authentication service');
+
+            securityService = require('./services/common/securityServiceOAuth2');
+        } else {
+            logger.info(context, 'Falling back to Keystone as authentication service');
```

Comment:
I agree that if authentication.type is not provided is a "Failling back" case, but if the user explicitely configured authentication.type === 'keystone' is a "Keystone selected case".

This problem could be avoided with a slight change in the log messages:

"Using OAuth2 as authentication service"
"Using Keystone as authentication service"

Reply:
fixed by b2682c7